### PR TITLE
(DOCSP-36281): Add new appservices deploy CLI docs

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Jira tickets: 
- https://jira.mongodb.org/browse/DOCSP-36281
- https://jira.mongodb.org/browse/DOCSP-35432
- https://jira.mongodb.org/browse/DOCSP-35672

Updated page:
- [Deploy Automatically with GitHub](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-36281/apps/deploy-github/): Add tabs for CLI and API procedures, update details about making changes from the CLI.

New pages (autogenerated from CLI docs):
- [appservices deploy](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-36281/cli/appservices-deploy/)
- [appservices deploy configure](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-36281/cli/appservices-deploy-configure/)
- [appservices deploy describe](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-36281/cli/appservices-deploy-describe/)
- [appservices deploy disable](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-36281/cli/appservices-deploy-disable/)
- [appservices deploy enable](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-36281/cli/appservices-deploy-enable/)

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-realm work, if any

### Release Notes

- **Develop & Deploy Apps**
  - Deploy Automatically with GitHub: Add tabs for CLI and API procedures, update details about making changes from the CLI.
- **App Services CLI**
  - appservices deploy: New section with details for the new CLI commands that enable you to manage automatic deployments of your app; `appservices deploy configure`, `appservices deploy describe`, `appservices deploy disable`, `appservices deploy enable`.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/apps/deploy-github.txt [hunk 1]
```diff
@@ -45,160 +51,269 @@ Prerequisites
...
+            :ref:`appservices-deploy-configure`.
 
-      When you are ready to deploy, commit all of the files that you want to include
-      and then push them to GitHub:
+            {+cli+} immediately enables automatic deployment when you complete
+            this command. You don't need to run the ``enable`` command as
+            an additional step.
 
-      .. code-block:: sh
+            .. tip:: This functionality relies on a GitHub beta feature
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:246]:
```suggestion
            .. tip:: Configuring automatic deployment via CLI relies on a GitHub beta feature
```

File: source/apps/deploy-github.txt [hunk 2]
```diff
@@ -45,160 +51,269 @@ Prerequisites
...
-         deployments in the :ref:`Deployment History
-         <deployment-history>` table in the App Services UI.
+            You can use the {+cli+} to view configuration details after you have
+            configured automatic deployment.
 
+            .. code-block:: shell
+
+               {+cli-bin+} deploy describe
+
+            You can disable and enable automatic deployment using the {+cli+}.
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:269]:
```suggestion
            You can also disable and enable automatic deployment using the {+cli+}. 
```

File: source/apps/deploy-github.txt [hunk 3]
```diff
@@ -45,160 +51,269 @@ Prerequisites
...
+               to configure automatic deployment using the {+cli+}. In this
+               case, use the App Services UI or Admin API to configure 
+               automatic deployment.
 
-      Once you successfully push your changes to GitHub, App Services
-      immediately deploys a new version of your application that matches the state
-      of the latest commit. Client applications will automatically use the newest
-      version, so make sure that you also update your client code to use the new
-      version if necessary.
+               This only applies to configuring automatic deployment. If you
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:255]: would it be accurate to say that this only applies to the ``appservices deploy configure`` command, and the ``deploy enable`` and ``deploy disable`` commands don't rely on the feature?

File: source/apps/deploy-github.txt [hunk 4]
```diff
@@ -45,160 +51,269 @@ Prerequisites
...
+
+            .. code-block:: bash
+
+               curl -X GET \
+                  https://services.cloud.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/deploy/config \
+                  -H 'Authorization: Bearer <access_token>'
+
+         .. step:: Update the Deployment Configuration
+
+            You can get the existing deployment configuration for your App by
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:297]:
```suggestion
            You can update the existing deployment configuration for your App by 
```

File: source/apps/deploy-github.txt [hunk 5]
```diff
@@ -233,11 +348,24 @@ bot appears as the commit author.
 
 .. _deploy-github-make-changes-cli:
 
-Avoid Making Changes from the CLI
----------------------------------
+Make Changes from the CLI
+-------------------------
+
+Previously, we have recommended avoiding avoid making changes to your app
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:354]:
```suggestion
Previously, we have recommended that you avoid making changes to your app 
```

File: source/apps/deploy-github.txt [hunk 6]
```diff
@@ -233,11 +348,24 @@ bot appears as the commit author.
...
 .. _deploy-github-make-changes-cli:
 
-Avoid Making Changes from the CLI
----------------------------------
+Make Changes from the CLI
+-------------------------
+
+Previously, we have recommended avoiding avoid making changes to your app 
+with {+cli+} after configuring automatic deployment. Starting with 
+{+cli+} v1.1.0 and newer, you can safely use the {+cli+} to make changes to
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:356]: had to double check but this is what we agreed on to use
```suggestion
{+cli+} v1.1.0 and later, you can safely use the {+cli+} to make changes to 
```

File: source/apps/deploy-github.txt [hunk 7]
```diff
@@ -233,11 +348,24 @@ bot appears as the commit author.
...
----------------------------------
+Make Changes from the CLI
+-------------------------
+
+Previously, we have recommended avoiding avoid making changes to your app 
+with {+cli+} after configuring automatic deployment. Starting with 
+{+cli+} v1.1.0 and newer, you can safely use the {+cli+} to make changes to 
+your app.
+
+**{+cli+} version 1.1.0 and newer**
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:359]:
```suggestion
**{+cli+} version 1.1.0 and later**
```

File: source/apps/deploy-github.txt [hunk 8]
```diff
@@ -233,11 +348,24 @@ bot appears as the commit author.
...
+
+In {+cli+} v1.1.0 and newer, a push from the {+cli+} automatically generates
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:361]:
```suggestion
In {+cli+} v1.1.0 and earlier, a push from the {+cli+} automatically generates
```

File: source/apps/deploy-github.txt [hunk 9]
```diff
@@ -233,11 +348,24 @@ bot appears as the commit author.
...
+a corresponding commit in GitHub. This keeps your code changes synchronized
+with your version-controlled GitHub source of truth.
+
+**{+cli+} version 1.0.3 and older**
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:365]:
```suggestion
**{+cli+} version 1.0.3 and earlier**
```

File: source/apps/deploy-github.txt [hunk 10]
```diff
@@ -233,11 +348,24 @@ bot appears as the commit author.
...
 
-With automatic GitHub deployment enabled, avoid making changes to your app 
-with {+cli+} because:
+In {+cli+} v1.0.3 and older, avoid making changes to your app with the {+cli+}
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:367]:
```suggestion
In {+cli+} v1.0.3 and earlier, avoid making changes to your app with the {+cli+}
```

File: source/apps/deploy-github.txt [hunk 11]
```diff
@@ -262,5 +390,5 @@ Summary
   GitHub Deployment, those changes are automatically committed back to your 
   linked repository.
 
-- Do not use {+cli+} to update configuration files after enabling 
-  Automatic GitHub Deployment.
+- If you would like to use {+cli+} to update configuration files after enabling 
+  Automatic GitHub Deployment, use {+cli+} v1.1.0 or newer.
```
Comment (cbullinger) [inline source/apps/deploy-github.txt:394]:
```suggestion
  Automatic GitHub Deployment, use {+cli+} v1.1.0 or later.
```

File: source/cli/appservices-deploy-configure.txt [hunk 12]
```diff
@@ -0,0 +1,87 @@
+.. _appservices-deploy-configure:
+
+============================
+appservices deploy configure
+============================
+
+.. default-domain:: mongodb
```
Comment (cbullinger) [inline source/cli/appservices-deploy-configure.txt:7]: are we doing metadata for these pages?

File: source/cli/appservices-deploy-configure.txt [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Configure the settings for automatic deployment
+
+Walks you through configuring automatic deployment for your app
```
Comment (cbullinger) [inline source/cli/appservices-deploy-configure.txt:17]: do we want to mention here or on the deploy ref landing page that the CLI automatically enables auto deployment when you complete the interactive configuration?

**Review by cbullinger — APPROVED** [review]
some non-blocking suggestions

File: source/cli/appservices-deploy-configure.txt (hunk 12 above)
Comment (dacharyc) [reply source/cli/appservices-deploy-configure.txt:7]: Nope! These pages are automatically generated, so we generally don't edit them. Any edits we make will be overwritten when they're regenerated.

File: source/cli/appservices-deploy-configure.txt (hunk 13 above)
Comment (dacharyc) [reply source/cli/appservices-deploy-configure.txt:17]: The content on these pages comes directly from the CLI code. Any changes we would make to these auto-generated pages would get overwritten. The engineering team would need to make that change in their codebase.

File: source/apps/deploy-github.txt (hunk 3 above)
Comment (dacharyc) [reply source/apps/deploy-github.txt]: Updating the verbiage to clarify details!

File: source/apps/deploy-github.txt [hunk 14]
```diff
@@ -45,160 +51,269 @@ Prerequisites
...
+
+            .. note:: GitHub Authentication
+
+               GitHub may require you to provide your GitHub account credentials before
+               you can install the app.
+
+            Select the GitHub account or organization for which you want to install the
+            app:
+
+            .. figure:: /images/github-app-install-location.png
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/apps/deploy-github.txt:83]: Sounds like a good discussion to have as a team! I'll resolve this comment here, and make a note for us to revisit process around images in procedures.

File: source/apps/deploy-github.txt [hunk 15]
```diff
@@ -233,11 +349,24 @@ bot appears as the commit author.
...
+-------------------------
+
+Previously, we have recommended that you avoid making changes to your app 
+with {+cli+} after configuring automatic deployment. Starting with 
+{+cli+} v1.1.0 and later, you can safely use the {+cli+} to make changes to 
+your app.
+
+**{+cli+} version 1.1.0 and later**
+
+In {+cli+} v1.1.0 and earlier, a push from the {+cli+} automatically generates
```
Comment (nlarew) [inline source/apps/deploy-github.txt:362]:
```suggestion
In {+cli+} v1.1.0 and later, a push from the {+cli+} automatically generates
```

**Pull request merged**